packages/malloc/src/pool.ts

```diff
@@ -10,42 +10,108 @@ import {
 } from "./api";
 import { wrap } from "./wrap";
 
+const POINTER_TOP_OFFSET = 0;
```

Comment:
Renaming these consts here is related to the comments below re: using `Uint32Array` instead of `DataView` to simplify:

```ts
const STATE_TOP = 0;
const STATE_END = 1;
const STATE_FREE = 2;
const STATE_USED = 3;

const STATE_NUM_FIELDS = 4;
const SIZEOF_STATE = STATE_NUM_FIELDS * Uint32Array.BYTES_PER_ELEMENT;
```